### DISCUSSION TITLE
push notification Icon

### DISCUSSION THREAD

**Discussion by oracle2000** [shinyorg/shiny#1501] [Q&A]
there is a way to choose notification icon?

**Comment by aritchie** [comment]
For iOS, you can use a notification extension.  That's outside of Shiny.
For Android, you can use this sample point: https://github.com/shinyorg/pushtester/blob/53022d8faf395e547db3f32bf0629e0254a39d2e/PushEndToEnd/Delegates/MyPushDelegate.cs#L27

This will give you full access to the notification object from push and allow you to change it as needed

**Comment by oracle2000** [comment]
Sorry but when i receive a notification in the allert on the phone the icon in a white square and not my icon

Sorry for my poor english

 

 

Fabio Falaschi

HelpDesk

Informatica Futura Srl 

Confidenziale. Questo messaggio di posta elettronica può contenere informazioni di carattere confidenziale rivolte esclusivamente al destinatario sopra indicato.Nel caso aveste ricevuto questo messaggio di posta elettronica per errore, siete pregati di segnalarlo immediatamente al mittente e distruggere quanto ricevuto (compresi i file allegati) senza farne copia. E' vietato l'uso, la diffusione, distribuzione o riproduzione da parte di ogni altra persona.Qualsivoglia utilizzo non autorizzato del contenuto di questo messaggio costituisce violazione dell'obbligo di non prendere cognizione della corrispondenza tra altri soggetti, salvo più grave illecito, ed espone il responsabile alle relative conseguenze.

Confidentially notice. This e-mail transmission may contain legally privileged and/or confidential information. Please do not read it if you are not the intended recipient(S).Any use, distribution, reproduction or disclosure by any other person is strictly prohibited.If you have received this e-mail in error, please notify the sender and destroy the original transmission and its attachments without reading or saving it in any manner.

 

Da: Allan Ritchie ***@***.***> 
Inviato: venerdì 12 luglio 2024 18:55
A: shinyorg/shiny ***@***.***>
Cc: oracle2000 ***@***.***>; Author ***@***.***>
Oggetto: Re: [shinyorg/shiny] push notification Icon (Discussion #1501)

 

For iOS, you can use a notification extension. That's outside of Shiny.
For Android, you can use this sample point: https://github.com/shinyorg/pushtester/blob/53022d8faf395e547db3f32bf0629e0254a39d2e/PushEndToEnd/Delegates/MyPushDelegate.cs#L27

This will give you full access to the notification object from push and allow you to change it as needed

**Reply by aritchie** [reply]
It means you haven't setup all of the required documents in your iOS app to cover that notification icon.  The sample app I mentioned, still has all of the answers you're looking for.  You just need to set your MAUI assets up.

**Comment by oracle2000** [comment]
Ok, my app is for android

That is my code:

        public Task OnReceived(PushNotification notification)

        {

 

#if ANDROID

        // TODO: if in foreground

 

        ((AndroidPushNotification)notification).Notify(100,((AndroidPushNotification)notification).CreateBuilder()

        .SetBadgeIconType(1)

        .SetSmallIcon(Resource.Drawable.notifica));

        

        

#endif

            MessagingCenter.Send<MainPage>((MainPage)Application.Current.MainPage, "notifica arrivata");

            return Task.CompletedTask;

 

        }

I attached my icon

Thanks a lot

 

 

Fabio Falaschi

HelpDesk

Informatica Futura Srl 

Confidenziale. Questo messaggio di posta elettronica può contenere informazioni di carattere confidenziale rivolte esclusivamente al destinatario sopra indicato.Nel caso aveste ricevuto questo messaggio di posta elettronica per errore, siete pregati di segnalarlo immediatamente al mittente e distruggere quanto ricevuto (compresi i file allegati) senza farne copia. E' vietato l'uso, la diffusione, distribuzione o riproduzione da parte di ogni altra persona.Qualsivoglia utilizzo non autorizzato del contenuto di questo messaggio costituisce violazione dell'obbligo di non prendere cognizione della corrispondenza tra altri soggetti, salvo più grave illecito, ed espone il responsabile alle relative conseguenze.

Confidentially notice. This e-mail transmission may contain legally privileged and/or confidential information. Please do not read it if you are not the intended recipient(S).Any use, distribution, reproduction or disclosure by any other person is strictly prohibited.If you have received this e-mail in error, please notify the sender and destroy the original transmission and its attachments without reading or saving it in any manner.

 

Da: Allan Ritchie ***@***.***> 
Inviato: lunedì 15 luglio 2024 19:35
A: shinyorg/shiny ***@***.***>
Cc: oracle2000 ***@***.***>; Author ***@***.***>
Oggetto: Re: [shinyorg/shiny] push notification Icon (Discussion #1501)

 

It means you haven't setup all of the required documents in your iOS app to cover that notification icon. The sample app I mentioned, still has all of the answers you're looking for. You just need to set your MAUI assets up.
